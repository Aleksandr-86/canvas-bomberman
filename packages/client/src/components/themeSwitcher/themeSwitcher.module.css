.switch {
  position: absolute;
  top: 18px;
  left: 25px;
  display: inline-block;
  width: 60px;
  height: 34px;
}

.switch input {
  width: 0;
  height: 0;
  opacity: 0;
}

.slider {
  position: absolute;
  top: 0;
  left: 0;
  width: 60px;
  height: 34px;
  background-color: #ccc;
  cursor: pointer;
  transition: 0.4s;
}

.slider::before {
  content: '';
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  width: 40px;
  height: 40px;
  margin: auto 0;
  background: white url('../../assets/images/lightIcon.png');
  background-position: center;
  background-repeat: no-repeat;
  box-shadow: 0 0 15px #2020203d;
  transition: 0.4s;
}

input:checked + .slider {
  background-color: #2196f3;
}

input:focus + .slider {
  box-shadow: 0 0 1px #2196f3;
}

input:checked + .slider::before {
  background: white url('../../assets/images/darkIcon.png');
  background-position: center;
  background-repeat: no-repeat;
  transform: translateX(24px);
}

.slider.round {
  border-radius: 34px;
}

.slider.round::before {
  border-radius: 50%;
}
