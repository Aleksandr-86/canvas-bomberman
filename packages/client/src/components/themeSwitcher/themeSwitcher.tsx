import React from 'react'
import { useAppDispatch, useAppSelector } from '../../store/hooks'
import { getTheme } from '../../store/selectors'
import { setTheme } from '../../store/themeSlice'
import styles from './themeSwitcher.module.css'

export const ThemeSwitcher: React.FC = () => {
  const dispatch = useAppDispatch()
  const theme = useAppSelector(getTheme)

  return (
    <label className={styles.switch}>
      <input
        type="checkbox"
        checked={theme === 'dark'}
        onChange={() => {
          if (theme === `light`) {
            document.documentElement.dataset.theme = `dark`
            dispatch(setTheme(`dark`))
          } else {
            document.documentElement.dataset.theme = `light`
            dispatch(setTheme(`light`))
          }
        }}
      />
      <span className={`${styles.slider} ${styles.round}`}></span>
    </label>
  )
}
