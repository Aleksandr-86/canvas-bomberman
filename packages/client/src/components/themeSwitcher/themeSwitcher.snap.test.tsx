import * as renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { ThemeSwitcher } from './themeSwitcher'

const mockStore = configureStore([])
const store = mockStore({
  user: {
    isAuth: true,
    user: {
      id: 1,
    },
  },
  theme: {
    current: `mockLight`,
  },
})

describe(`<ThemeSwitcher/> snapshot test cases`, () => {
  test(`<ThemeSwitcher/> should render <label> parent`, () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ThemeSwitcher />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
