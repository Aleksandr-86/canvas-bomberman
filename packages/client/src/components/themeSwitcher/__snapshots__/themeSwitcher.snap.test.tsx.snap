// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemeSwitcher/> snapshot test cases <ThemeSwitcher/> should render <label> parent 1`] = `
<label
  className="switch"
>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="slider round"
  />
</label>
`;
